tlvc/arch/i86/boot/setup.S

```diff
@@ -514,13 +515,21 @@ sys_hdr_good:
 	xor %di,%di
 
 #ifdef CONFIG_ARCH_IBMPC
-// Check whether relocating .text to HMA is possible
+// If 80286 or higher, check whether relocating .text to HMA is possible
+	push %ds
+	mov $INITSEG,%ax
+	mov %ax,%ds
+	mov cpu_type,%ax
+	cmpb $6,cpu_type
+	jl 2f
```

Comment:
This code should be moved to checkhma, shouldn't it? The sole purpose of checkhma is to determine whether HMA boot is a viable option or not. This would also eliminate the kluge DS push since it (previously) always referenced DS==INITSEG.